### ISSUE TITLE
[edn/rtl/dv] EDN handle cmd ready de-assert

### ISSUE THREAD

**Issue by senelson7** [lowRISC/opentitan#15561]
- PR #15478 Contains a fifo-based fix to allow block-level/chip-level tests to pass.
- Once this PR is merged, submit block-based edn/dv PR to randomly create this condition, Tests will include edn_genbits, edn_smoke.
- Need to create new PR that removes additional fifo and fixes handshaking in all modes.

**Comment by senelson7** [comment]
dv PR #15570

**Comment by mwbranstad** [comment]
The above mentioned first two steps have now been completed. Here is an update on the EDN RTL after some consideration:
The ready issue appears to only affect "auto-request" mode, which is expected to be used by the EDN instance connected to the OTBN complex. The "boot-mode" is not affected because only single beat requests are made, and completed with a CSRNG ack for each command. Likewise, "software" mode is also unaffected because it does look at the ready input signal. Because of this, the software mode could have been used as a work-around, but would defeat the convenience of the RTL auto-request mode function.
The fix for the "cost-reduced" function could be done by the following changes:
- remove the output FIFO, and related alert.
- change the output flop stage to hold state when the CSRNG cmd ready input is low.
- the ready input most likely needs to be monitored by the main state machine, since the last beat of a command was not behaving properly with the mostly functioning fix that @vogelpi put together.
- there may be some other cases to close, but the unit tb will now expose them all with enough simulation runs.

**Comment by vogelpi** [comment]
This is something we should still consider for D3. The required DV components are now there. Removing these additional FIFOs would save 5 - 6.5 kGE for ES.

**Comment by tjaychen** [comment]
yeah agreed, especially if such a fix does not cause existing DV churn.

**Comment by tjaychen** [comment]
I'm movoing this to M3 for now based on the priority definition @msfschaffner is working on. 
That feels appropriate to me.

**Comment by vogelpi** [comment]
Hey @tjaychen , based on the M3 triaging guidelines I think this should be a P3 (nice to have but manageable without). Please let me know if you don't agree.

**Comment by tjaychen** [comment]
i think so too.

**Comment by andreaskurth** [comment]
Triaged for `edn`. Assigning to M2.5 with https://github.com/lowRISC/opentitan/labels/Priority%3AP3 under the "nice to have but manageable without" rationale.

**Comment by msfschaffner** [comment]
Planned for PROD as P1, since fixing this the right way will save several kGE and the DV is already in place for this.
